packages/gatsby-source-contentful/src/fetch.js

```diff
@@ -4,14 +4,81 @@ const chalk = require(`chalk`)
 const { formatPluginOptionsForCLI } = require(`./plugin-options`)
 const { CODES } = require(`./report`)
 
+/**
+ * Generate a user friendly error message.
+ *
+ * Contentful's API has its own error message structure, which might change depending of internal server or authentification errors.
+ *
+ * Additionally the SDK strips the error object, sometimes:
+ * https://github.com/contentful/contentful.js/blob/b67b77ac8c919c4ec39203f8cac2043854ab0014/lib/create-contentful-api.js#L89-L99
+ *
+ * This code tries to work around this.
+ */
+const createContentfulErrorMessage = e => {
+  // If we got a response, it is very likely that it is a Contentful API error.
+  if (e.response) {
+    let parsedContentfulErrorData = {}
+
+    // Parse JSON response data, and add it to the object.
+    if (typeof e.response.data === `string`) {
+      try {
+        parsedContentfulErrorData = JSON.parse(e.response.data)
+        if (typeof parsedContentfulErrorData !== `object`) {
+          throw new Error()
+        }
```

Comment:
JSON.parse('string') will throw an error when it's not parsable. Unsure what you're trying to guard against.